### ISSUE TITLE
Random log out back to login screen

### ISSUE THREAD

**Issue by meriops** [swaywm/sway#8279]
- **Sway Version:** 1.10-dev-4d4c88f0 (Jul 23 2024, branch 'master')

- **Debug Log:** [sway.log](https://gist.github.com/meriops/cd0f5b64a6f0addb99923e6a870c0ae8)

- **Configuration File:** [config file](https://gist.github.com/meriops/7fdd44ebe080ce9ab44b1cf54f222017)

- **Stack Trace:** [stack trace](https://gist.github.com/meriops/3c10b6baea0814556c85c21711120e8e)

- **Description:**
Basically, sway randomly logs me out of the session and displays the login screen (greetd + greetd-tuigreet). I cannot reproduce in a consistent way, as it really seems random with respect to both when and how often it happens. The sway debug file indicates an error mentioning wezterm, but the crash happens too with foot or alacritty as the only running terminal. It does not happen when I ssh to the machine, which makes me think it is sway / wlroots related. 
It might as well be something I messed up somewhere in a configuration file since I have a similar configuration on two other machines (desktop, laptop) with different hardware, where the crash does not happen (although it did happen at some point on the laptop, but 'went away' after a while...).
I would greatly appreciate some pointers to try and figure out what I'm doing wrong here, since this crash really hinders my work...  Please let me know where to post this if here is not the right place.

**Comment by emersion** [comment]
The backtrace you provided doesn't contain debug symbols. This most likely happens because the Sway binary you're using doesn't have debug information bundled.

Can you try again with a manually compiled Sway binary? See https://github.com/swaywm/sway/wiki/Development-Setup#compiling-as-a-subproject

**Comment by meriops** [comment]
Thank your for your reply. I am in a busy period right now, so it will likely take some days/weeks before I can get back to you with the proper debug information.

**Comment by meriops** [comment]
I managed to try with a manually compiled sway... I added [here](https://gist.github.com/meriops/ee58f8a3a33891fc61a2aee8ebce0615) the log file, the coredump, and the output of gdb... hopefully, there's enough information now. Please let me know if it is not the case. Thanks again for your help!

**Comment by emersion** [comment]
Sounds like this a nouveau bug:

```

nouveau: kernel rejected pushbuf: No such device
nouveau: ch16: krec 0 pushes 1 bufs 4 relocs 0
nouveau: ch16: buf 00000000 0000000e 00000004 00000004 00000000 0x7b946e870000 0x7f0000 0x80000
nouveau: ch16: buf 00000001 00000006 00000004 00000000 00000004 0x7b94835fa000 0x21c000 0x1000
nouveau: ch16: buf 00000002 0000002c 00000002 00000000 00000002 (nil) 0xf21000 0x1000
nouveau: ch16: buf 00000003 0000002e 00000004 00000004 00000000 0x7b94835d9000 0xf22000 0x1000
nouveau: ch16: psh 00000000 000002a214 000002a264
nouveau: 	0x200181c2
nouveau: 	0x03303210
nouveau: 	0x200681c3
nouveau: 	0x00001020
nouveau: 	0x00000018
nouveau: 	0x00000018
nouveau: 	0x00000001
nouveau: 	0x00000000
nouveau: 	0x00000000
nouveau: 	0x20088100
nouveau: 	0x00000000
nouveau: 	0x00f22000
nouveau: 	0x00000000
nouveau: 	0x00f21000
nouveau: 	0x00000060
nouveau: 	0x00000080
nouveau: 	0x00000018
nouveau: 	0x00000018
nouveau: 	0x200180c0
nouveau: 	0x00000686
nouveau: kernel rejected pushbuf: No such device
nouveau: ch16: krec 0 pushes 1 bufs 0 relocs 0
```

**Comment by BlakeOrth** [comment]
I had initially thought @atk91's mouse cursor theory was a bit far fetched, but moving my mouse in a way that caused rapid cursor changes indeed caused a crash with this signature relatively quickly.

If anyone else is running into this issue, I had been crashing at least daily, and certain web tools would often crash within minutes of usage. The workaround seems to be setting `WLR_NO_HARDWARE_CURSORS=1`, after which I have yet to see a crash. If I end up crashing with the variable set I'll update this comment to adjust expectations appropriately.

Sway+nouveau info for posterity:
```
local/linux 6.11.1.arch1-1
local/mesa 1:24.2.3-1
local/sway 1:1.9-5
```

**Comment by alexander-belikov** [comment]
It happens to me too. I'll try @BlakeOrth 's suggestion and report.

**Comment by alexander-belikov** [comment]
No, still getting logged out with `WLR_NO_HARDWARE_CURSORS=1` :/

**Comment by BlakeOrth** [comment]
@alexander-belikov thanks for tagging me here, I had forgotten to update my comment with my experience after using the variable for some time. I've edited the comment to my experience, which is perhaps not dissimilar from yours.

**Comment by wasowski** [comment]
I am seeing very similar things.  It does seem that putting the mouse away reduces the frequence.  For now I have set `WLR_NO_HARDWARE_CURSORS=1` and have not seen a crash for 24h (despite using the mouse).

**Comment by gdeoliveira** [comment]
I also seem to be experiencing this issue after a fresh install in a new laptop (I haven't experienced it before on a previous laptop where I've had Sway installed for some time).
What I've noticed is that the issue seems to happen only (or at least much more frequently) when I'm using the laptop's touch pad, but not as much (maybe never?) when I'm using a USB mouse plugged into the laptop instead.

**Comment by redrifle** [comment]
I am also getting this issue with the same output emersion posted. On a fedora 42 I just installed an hour ago. RTX 3070. Except I am using labwc.

**Comment by kosumic** [comment]
I am having the same issue after a recent upgrade.
sway version 1.11 on an AMD machine. 

I tried to run `sway` with `WLR_RENDERER=vulkan sway`, but it still happens.

I don't use a mouse but I notice the touchpad seems to be triggering the logout

**Comment by Charles-Schleich** [comment]
Adding here that in my case,
` WLR_NO_HARDWARE_CURSORS = "1";`
has made my system stable going on 3 days of up time now.

```bash
λ uname -a
Linux nixos 6.12.53 #1-NixOS SMP PREEMPT_DYNAMIC Wed Oct 15 10:00:25 UTC 2025 x86_64 GNU/Linux
```

```bash
λ nixos-version
25.05.20251019.33c6dca (Warbler)
```

```bash
λ sway --version
sway version 1.11
```

I did not need to set
`WLR_RENDERER=vulkan sway`

Happy to provide more context if needed.

**Comment by akumria** [comment]
Not a Sway user, but I've been chasing the identical fault (`nouveau: kernel rejected pushbuf: No such device`) on GNOME/Mutter and wanted to share data since it seems to rule out the wlroots-cursor theory as the sole cause.

**Hardware:** NVIDIA GK208B [GeForce GT 730] (Kepler), driven by `nouveau`.

**Software:** Debian testing. Reproduced identically on two kernels ( `7.1.8+deb14.1-amd64` and `7.1.12+deb14-amd64`). Mesa 26.1.6, GNOME Shell/Mutter 50.3 and 50.4 (also reproduced across a Mutter version bump).

**Trigger:** No wlroots/hardware-cursor involvement at all in my case. Using a native-Wayland GNOME Shell session. I've hit the exact same fault three times in ~4 days, triggered by two browsers under normal use:

1. Google Chrome (152.0.7977.82) - GPU submission on gnome-shell's own compositor channel got rejected, which took down the entire Wayland session and X server, forcing a new session from the GDM login screen. Reproduced twice this way.
2. Firefox (155.0~build1)- same kernel message. this time Firefox generated a clean minidump and exited, and the desktop session survived.

Whether the fault kills just the offending client's channel or cascades into killing the compositor's own channel (and the whole session) looks driver-side, not compositor-side.

One more data point: `vulkaninfo` enumerates this GPU under Mesa's NVK driver (`GeForce GT 730 (NVK GK208B)`, Vulkan 1.2), so the newer kernel-uAPI submission path seems fine. This fault only appears to hit the classic/legacy nouveau kernel DRM submission path that GL clients (Mutter, Chrome GL, Firefox GL) use.

Let me know if if the  full `nouveau: ch*` pushbuf dumps would be useful.

**Comment by kubahanslik** [comment]
Adding this in case someone makes the same mistake as I did:
From my experience, it seems to be important to export the variable `WLR_NO_HARDWARE_CURSORS = 1` before actually launching sway. Ideally edit your `/usr/share/wayland-sessions/sway.desktop`.

Haven't crashed since.

EDIT:
2 weeks and still no crash.
